FiguresInheritance/Program.cs

```diff
@@ -0,0 +1,45 @@
+﻿using FiguresInheritance.domain;
+using FiguresInheritance.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace FiguresInheritance
+{
+    internal class Program
+    {
+        static void Main(string[] args)
+        {
+            // Единственное, нет проверок на отрицательные числа и на стороны (потипу как для треугольника)
```

Comment:
If I hadn't seen this comment, I wouldn't have said anything. BUT if you insist. I'm pretty sure that you would like to implement those types of checks for me. You should do this logic inside setters

Reply:
Now there is